Keypad/Keypad.js

```diff
@@ -0,0 +1,264 @@
+							subtext={keyText.subtext}
+						>
+							{keyText.text === 'arrowleftturn' || keyText.text === 'phone' ? keyText.text : null}
+						</Cell>
+					);
+				})}
+			</Layout>
+		);
+	}
+});
+
+/**
+ * A Keypad component with an Input to display the outcome.
+ *
+ * @class Keypad
+ * @memberof agate/Keypad
+ * @extends agate/Keypad.KeypadBase
+ * @mixes agate/Input
+ * @ui
+ * @public
+ */
+class Keypad extends React.Component {
+	static propTypes = /** @lends agate/Keypad.prototype */ {
+		disabled: PropTypes.bool,
+		handleInputValue: PropTypes.func,
+		value: PropTypes.string
+	}
+
+	constructor (props) {
+		super(props);
+		this.state = {
+			keypadInput: '',
+			charIndex: 0
+		};
+	}
+
+	getCharIndex = (e) => {
+		this.setState({
+			charIndex: e.target.selectionStart
+		});
+	}
+
+	handleInputValue = (keyValue) => {
+		const {keypadInput, charIndex} = this.state;
+		let newKeypadInput = keypadInput;
+		let newCharIndex;
+
+		switch (keyValue) {
+			case 'arrowleftturn':
+			case 'Backspace':
+				newCharIndex = charIndex;
+				newKeypadInput = newKeypadInput.substring(0, charIndex - 1) + newKeypadInput.substring(charIndex, newKeypadInput.length);
+				newCharIndex = newCharIndex - 1;
+
+				this.setState({
+					charIndex: newCharIndex
+				});
+				break;
+
+			case 'ArrowLeft':
+				if (charIndex >= 0) {
+					newCharIndex = charIndex;
+					newCharIndex = newCharIndex - 1;
+
+					this.setState({
+						charIndex: newCharIndex
+					});
+				}
+				break;
+
+			case 'ArrowRight':
+				newCharIndex = charIndex;
+				newCharIndex = newCharIndex + 1;
+
+				this.setState({
+					charIndex: newCharIndex
+				});
+				break;
+
+			case 'Delete':
+				newCharIndex = charIndex;
+				newKeypadInput = newKeypadInput.substring(0, charIndex) + newKeypadInput.substring(charIndex + 1, newKeypadInput.length);
+				newCharIndex = newCharIndex - 1;
+
+				this.setState({
+					charIndex: newCharIndex
+				});
+				break;
+
+			case 'ArrowUp':
+			case 'ArrowDown':
+				// do nothing;
+				break;
+
+			case 'phone':
+				// method to call dialed number (keypadInput);
+				break;
+
+			default:
+				newCharIndex = charIndex;
+				newKeypadInput = newKeypadInput.substring(0, charIndex) + keyValue + newKeypadInput.substring(charIndex);
+				newCharIndex = newCharIndex + 1;
+
+				this.setState({
+					charIndex: newCharIndex
+				});
+				break;
+		}
+
+		this.setState({
+			keypadInput: newKeypadInput
+		});
+	};
+
+	render () {
+		const {handleInputValue, getCharIndex} = this,
+			{disabled} = this.props,
+			{keypadInput} = this.state;
+
+		return (
+			<React.Fragment>
+				<KeypadBase handleInputValue={handleInputValue} disabled={disabled} />
+				<Input
+					className={css.keypadInput}
+					css={css}
+					onClick={getCharIndex}
+					onKeyDown={(e) => handleInputValue(e.key)}
+					onKeyUp={getCharIndex}
+					type="tel"
+					value={keypadInput}
+				/>
+			</React.Fragment>
+		);
+	}
+}
+
+export default Keypad;
+export {
+	KeypadBase
```

Comment:
By default, it would be better to export `Keypad` and `KeypadBase` together.
If the `KeypadBase` is not needed out of the `Keypad`, let's remove it.

Reply:
```suggestion
	Keypad,
	KeypadBase
```